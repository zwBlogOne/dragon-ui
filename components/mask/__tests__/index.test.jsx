import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Mask from '../index';

describe('Mask', () => {
  it('renders different types of Mask correctly', () => {
    const wrapper = render(
      <div>
        <Mask type="transparent" visible />
        <Mask type="light" visible />
        <Mask type="dark" visible />
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders hidden Mask correctly', () => {
    const wrapper = render(
      <div>
        <Mask />
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('behaves correctly when closing', () => {
    const onClose = jest.fn();
    const wrapper = mount(
      <Mask visible onClose={onClose} />
    );

    wrapper.find('.ui-mask').simulate('click');

    expect(onClose).toHaveBeenCalled();
  });
});
