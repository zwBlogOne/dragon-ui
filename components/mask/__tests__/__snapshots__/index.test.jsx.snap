// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mask renders different types of Mask correctly 1`] = `
<div>
  <div
    class="ui-mask transparent"
  />
  <div
    class="ui-mask light"
  />
  <div
    class="ui-mask dark"
  />
</div>
`;

exports[`Mask renders hidden Mask correctly 1`] = `<div />`;
